zyngine/zynthian_layer.py

```diff
@@ -395,8 +395,16 @@ def midi_unlearn(self):
 
 
 	def midi_control_change(self, chan, ccnum, ccval):
-		if self.engine:
-			#logging.debug("Receving MIDI CH{}#CC{}={}".format(chan, ccnum, ccval))
+		if ccnum==0:
+			logging.debug("Setting Bank: Receiving MIDI CH{}#CC{}={}".format(chan, ccnum, ccval))
+			self.set_bank(ccval)
+			self.load_preset_list()
+		elif ccnum==32:
+			#since preset is the same as program change, lsb needs to be ignored
+			#logging.debug("Setting Preset: Receiving MIDI CH{}#CC{}={}".format(chan, ccnum, ccval))
+			pass
+		elif self.engine:
+			#logging.debug("Receiving MIDI CH{}#CC{}={}".format(chan, ccnum, ccval))
```

Comment:
I propose this behaviour:
+ When CC0 (MSB) is received, we save it to a flag var "bank_msb". 
+ When CC32 (LSB) is received, we change the bank, using the "bank_msb" like this:

    + bank_msb==0 => system banks LSB (0-127)
    + bank_msb==1 => user bank LSB (0-127)
    + bank_msb==2 => favourites bank, if it exist. In this case, lsb has not meaning.

Of course, other schemes are possible, but i find this very simple and  easy to remember/understand.